### PR TITLE
feat(mcp): add URL parameter binding for HTTP transport

### PR DESCRIPTION
**Pull request by duwenxin99**
This PR implements URL Parameter Binding for the HTTP transport. 

- Filters out connection-level URL parameters from the client-facing schemas during tools/list and automatically injects them during tools/call.
- Performs automatic type conversion of incoming string query parameters to simple types (integer, boolean, number) and unmarshals complex types (array, object) from JSON-encoded values.

### CODE REVIEW

File: internal/tools/tools.go
Comment (Yuan325) [inline internal/tools/tools.go:1]: This is updated. The McpManifest is now only generated when list method is called (tools list or pormpts list). It will be generated as needed, so we probbaly will not need to clone it anymore. 

The manifest generation code live in `internal/server/mcp/<version>/manifest.go`.

File: internal/server/mcp/v20251125/method.go [hunk 1]
```diff
@@ -157,6 +164,39 @@ func toolsCallHandler(ctx context.Context, id jsonrpc.RequestId, resourceMgr *re
 		return jsonrpc.NewError(id, jsonrpc.INTERNAL_ERROR, err.Error(), nil), err
 	}
 
+	// Auto-populate arguments from URL parameters
+	urlParams, ok := util.UrlParamsFromContext(ctx)
```
Comment (Yuan325) [inline internal/server/mcp/v20251125/method.go]: Should we add a function in `internal/util/parameters.go` for this?

File: docs/en/documentation/configuration/tools/url_parameter_binding.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+title: "URL Parameter Binding"
+type: docs
+weight: 10
+description: >
+  How to bind tool arguments at the transport level using URL query parameters.
+---
+
+## About
+
+URL Parameter Binding is a transport-level feature for the SSE (HTTP) transport that allows you to bind specific arguments to tools at the connection level. This is useful for creating scoped connections for generic MCP clients where you want to restrict the client to a specific database, project, or instance without hardcoding these values in the server configuration for all users, or requiring the client to provide them.
```
Comment (Yuan325) [inline docs/en/documentation/configuration/tools/url_parameter_binding.md]: This is not specific to SSE right?

File: docs/en/documentation/configuration/tools/url_parameter_binding.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+## How It Works
+
+When an MCP client connects to the server via SSE with query parameters (e.g., `http://localhost:5000/mcp/sse?project=my-project`):
```
Comment (Yuan325) [inline docs/en/documentation/configuration/tools/url_parameter_binding.md]: Is it just with SSE? Seems like we are making it work for all http transport in general (not sse specific).

**Review by Yuan325 — COMMENTED** [review]
i'll take another look once the conflicts are resolved. Thank you! :)

File: internal/server/mcp.go [hunk 4]
```diff
@@ -464,6 +468,20 @@ func httpHandler(s *Server, w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 	ctx = util.WithLogger(ctx, s.logger)
 
+	queryParams := r.URL.Query()
```
Comment (Yuan325) [inline internal/server/mcp.go:498]: Since this is within the `httpHandler()`, it will apply URL params for all http transport, not just SSE.

File: internal/server/mcp/v20251125/method.go (hunk 1 above)
Comment (duwenxin99) [reply internal/server/mcp/v20251125/method.go]: It's fetching from context so I think it would be better to put it together with the rest of the functions fetching info from context (such as `InstrumentationFromContext` and `GenAIMetricAttrsFromContext`)  in `internal/util/util.go`.

File: internal/server/mcp.go (hunk 4 above)
Comment (duwenxin99) [reply internal/server/mcp.go:498]: Sorry, miscommunicated previously. It is for all HTTP transport and not just SSE. Updated the PR title.

File: docs/en/documentation/configuration/tools/url_parameter_binding.md (hunk 3 above)
Comment (duwenxin99) [reply docs/en/documentation/configuration/tools/url_parameter_binding.md]: Right. Sorry for the misinformation previously. It is for all http transport.

File: internal/tools/tools.go
Comment (duwenxin99) [reply internal/tools/tools.go:1]: Updated.

File: docs/en/documentation/configuration/tools/url_parameter_binding.md (hunk 2 above)
Comment (duwenxin99) [reply docs/en/documentation/configuration/tools/url_parameter_binding.md]: No it is not.

File: docs/en/documentation/configuration/tools/url_parameter_binding.md (hunk 3 above)
Comment (averikitsch) [reply docs/en/documentation/configuration/tools/url_parameter_binding.md]: Can we update the wording here?

File: internal/server/mcp/v20241105/manifests.go [hunk 5]
```diff
@@ -68,6 +68,13 @@ func generateParamManifest(ps parameters.Parameters) (InputSchema, map[string][]
 		}
 
 		name := p.GetName()
+		if urlParams != nil {
```
Comment (averikitsch) [inline internal/server/mcp/v20241105/manifests.go:71]: Do we have any helper logs for 
1) same param name but different type 
2) params that are in URL but aren't used?

**Review by averikitsch — APPROVED** [review]

File: internal/server/mcp/util/url_binding.go [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+	"encoding/json"
+	"strconv"
+
+	"github.com/googleapis/mcp-toolbox/internal/util"
+	"github.com/googleapis/mcp-toolbox/internal/util/parameters"
+)
+
+// PopulateUrlParams injects bound URL query parameters into the data arguments
+// and performs automatic type conversion for integer, boolean, and float parameters.
+func PopulateUrlParams(ctx context.Context, data map[string]any, toolParams parameters.Parameters) map[string]any {
```
Comment (Yuan325) [inline internal/server/mcp/util/url_binding.go:28]: can we add unit tests to this function?

File: internal/server/mcp/v20241105/manifests_test.go [hunk 7]
```diff
@@ -188,7 +188,7 @@ func TestParamManifest(t *testing.T) {
 	}
 	for _, tc := range tcs {
 		t.Run(tc.name, func(t *testing.T) {
-			gotSchema, gotAuthParam := generateParamManifest(tc.in)
+			gotSchema, gotAuthParam := generateParamManifest(tc.in, nil)
```
Comment (Yuan325) [inline internal/server/mcp/v20241105/manifests_test.go]: can we add a test case here where the `urlParams` is not nil?

**Review by Yuan325 — APPROVED** [review]

File: docs/en/documentation/configuration/tools/url_parameter_binding.md (hunk 3 above)
Comment (duwenxin99) [reply docs/en/documentation/configuration/tools/url_parameter_binding.md]: updated
